File: quantum-vulnerable/quantum_vulnerable.c
// Quantum-vulnerable (Shor-breakable) public-key algorithms
#define ALG_P384_0 "P-384"
#define ALG_RSA768_1 "RSA-768"
#define ALG_STATICDH_2 "static-DH"
#define ALG_NACCACHESTERN_3 "Naccache-Stern"
#define ALG_BENALOH_4 "Benaloh"
#define ALG_SECP521R1_5 "secp521r1"
#define ALG_SECP256R1_6 "secp256r1"
#define ALG_RSA512_7 "RSA-512"
#define ALG_SECT571R1_8 "sect571r1"
#define ALG_ECGDSA_9 "ECGDSA"
#define ALG_NISTP256_10 "NIST-P256"
#define ALG_ECKCDSA_11 "ECKCDSA"
#define ALG_ED448_12 "Ed448"
#define ALG_MODP4096_13 "MODP-4096"
// TODO migrate P-521 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate NIST-P256 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate sect283k1 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate P-256 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate KCDSA to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate P-384 to a post-quantum scheme (harvest-now-decrypt-later)
const char *enabledAlgorithms = {"ECGDSA", "MODP-1536", "ECDHE", "ElGamal", "DSA-1024", "ffdhe4096", "EC-GOST", "ffdhe3072"};

File: quantum-vulnerable/quantum_vulnerable.rs
// Quantum-vulnerable (Shor-breakable) public-key algorithms
const ALG_X942DH_0: &str = "X9.42-DH";
const ALG_ECKCDSA_1: &str = "ECKCDSA";
const ALG_ELGAMAL_2: &str = "ElGamal";
const ALG_ECSCHNORR_3: &str = "EC-Schnorr";
const ALG_BRAINPOOLP512R1_4: &str = "brainpoolP512r1";
const ALG_FFDHE4096_5: &str = "ffdhe4096";
const ALG_MODP8192_6: &str = "MODP-8192";
const ALG_SCHNORR_7: &str = "Schnorr";
const ALG_PAILLIER_8: &str = "Paillier";
const ALG_FFDHE8192_9: &str = "ffdhe8192";
const ALG_NISTP256_10: &str = "NIST-P256";
const ALG_OAKLEYGROUP14_11: &str = "Oakley-Group-14";
const ALG_RSAOAEP256_12: &str = "RSA-OAEP-256";
const ALG_SECT283K1_13: &str = "sect283k1";
// TODO migrate P-256 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate DSA-3072 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate Schnorr to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate EC-Schnorr to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate RSA-4096 to a post-quantum scheme (harvest-now-decrypt-later)
// TODO migrate ffdhe8192 to a post-quantum scheme (harvest-now-decrypt-later)
let enabledAlgorithms = vec!["BN254", "Okamoto-Uchiyama", "DSA", "X448", "X9.42-DH", "Curve25519", "Boneh-Lynn-Shacham", "BLS"];
